### DISCUSSION TITLE
Blank screen when uploading image

### DISCUSSION THREAD

**Discussion by justijndepover** [awcodes/filament-curator#336] [Q&A]
Hi

I have installed Curator in Filament (both on latest version).
When I try to upload an image in the Media module, I get a blank screen with no error.

Looking at the PHP logs, I see this:
[01-Nov-2023 15:13:56 Europe/Brussels] PHP Fatal error:  Allowed memory size of 134217728 bytes exhausted (tried to allocate 24576 bytes) in /Users/justijndepover/Sites/project-x/vendor/intervention/image/src/Intervention/Image/Gd/Decoder.php on line 38

It looks like this is related to this issue:
https://github.com/thephpleague/glide/issues/79

I tried increasing the memory limit first, but the problem remained.
Next I tried to add a custom ServerFactory to use imagick like so:

```php
<?php

namespace App\Filament\Curator;

use Awcodes\Curator\Glide\Contracts\ServerFactory as ContractsServerFactory;
use League\Glide\Responses\LaravelResponseFactory;
use League\Glide\Server;
use League\Glide\ServerFactory;

class CustomServerFactory implements ContractsServerFactory
{
    public function getFactory(): ServerFactory | Server
    {
        return ServerFactory::create([
            'driver' => 'imagick',
            'response' => new LaravelResponseFactory(app('request')),
            'source' => storage_path('app'),
            'source_path_prefix' => 'public',
            'cache' => storage_path('app'),
            'cache_path_prefix' => '.cache',
            'max_image_size' => 20000 * 20000,
        ]);
    }
}
```

And changed the config:
```php
'glide' => [
    'server' => \App\Filament\Curator\CustomServerFactory::class,
    // 'server' => \Awcodes\Curator\Glide\DefaultServerFactory::class,
    'fallbacks' => [],
],
```

But the memory issue still happens in this file:
```
/vendor/intervention/image/src/Intervention/Image/Gd/Decoder.php
```

I also checked with `extension_loaded('imagick')` and the imagick extension is loaded.
Does anyone know why I'm not able to use Imagick?

**Comment by awcodes** [comment]
Not sure, that all looks ok to me and I know other users who are using Imagick successfully.

**Reply by justijndepover** [reply]
I ended up publishing the intervention config and changing the driver there. That seemed to do the trick
